### ISSUE TITLE
`put 1 &` crashes Elvish

### ISSUE THREAD

**Issue by simonrouse9461** [elves/elvish#1362]
`put 1 &` will crash Elvish with the following error:
```
panic: send on closed channel

goroutine 152 [running]:
github.com/elves/elvish/pkg/eval.put(0xc0000c4f50, 0xc00041e0b0, 0x1, 0x1)
	github.com/elves/elvish/pkg/eval/builtin_fn_io.go:90 +0x93
reflect.Value.call(0x1418020, 0x14d1538, 0x13, 0x14ab823, 0x4, 0xc00041c1e0, 0x2, 0x2, 0x2, 0x18, ...)
	reflect/value.go:476 +0x8c7
reflect.Value.Call(0x1418020, 0x14d1538, 0x13, 0xc00041c1b0, 0x2, 0x2, 0x1, 0x2, 0xc000024d68)
	reflect/value.go:337 +0xb9
github.com/elves/elvish/pkg/eval.(*goFn).Call(0xc00014d810, 0xc0000c4f50, 0xc00041e080, 0x1, 0x1, 0xc00041c180, 0x0, 0x1)
	github.com/elves/elvish/pkg/eval/go_fn.go:237 +0x658
github.com/elves/elvish/pkg/eval.(*formOp).exec(0xc0000b4840, 0xc0000c4f50, 0x0, 0x0)
	github.com/elves/elvish/pkg/eval/compile_effect.go:375 +0x99d
github.com/elves/elvish/pkg/eval.(*pipelineOp).exec.func1(0x152f360, 0xc0000b4840, 0xc0000c4f50, 0xc0001903a0, 0xc000024d80, 0x203000)
	github.com/elves/elvish/pkg/eval/compile_effect.go:116 +0x3f
created by github.com/elves/elvish/pkg/eval.(*pipelineOp).exec
	github.com/elves/elvish/pkg/eval/compile_effect.go:115 +0x228
```

**Comment by simonrouse9461** [comment]
It doesn't always crash. Sometimes it's working, which gives this:
```
~> put 2 &
job put 2 & finished
▶ 2
~>
```

**Comment by zzamboni** [comment]
Since Elvish is in constant development, you should always indicate the version of Elvish with which you found the problem. I cannot reproduce this, even after multiple tries. Which version of Elvish are you using?

Mine:
```
|> elvish -buildinfo
Version: 0.16.0-dev.ebf02e3a9192d43ecd0b67f4ddbdb5230bd6e8d5
Go version: go1.16.5
Reproducible build: true
```

**Comment by simonrouse9461** [comment]
Here's the version I have:
```
Version: 0.15.0
Go version: go1.15.7
Reproducible build: false
```

**Comment by krader1961** [comment]
Dup of #1069 which I closed five days ago because a recent change fixed this problem.

**Comment by zzamboni** [comment]
@simonrouse9461 0.15.0 was released more than 6 months ago. As @krader1961 mentioned, this bug has already been fixed in HEAD.

**Comment by xiaq** [comment]
Hmm, I should probably have released a patch version to 0.15.0 for this fix. But since 0.16.0 will be released this month anyway, I'll skip that.

**Comment by xiaq** [comment]
Closing this now, please reopen if you can reproduce this in HEAD.
